---
title: 'Project Data Overview'
subtitle: 'Homework 2'
author: 
- Louis Coussement
- Menno van Damme
date: 2022-2023
output:
  html_document:
    df_print: paged
    toc: true
  pdf_document:
    dev: png
    toc: true
header-includes: \setlength{\emergencystretch}{3em}
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.align = 'center')
knitr::opts_chunk$set(out.width = '60%')
knitr::opts_chunk$set(fig.width=12, fig.height=8)

library("ArrayExpress")
library("arrayQualityMetrics")
library("ggplot2")
library("limma")
library("oligo")
library("siggenes")
library("affy")
library("pd.huex.1.0.st.v2")
library("RSQLite")
library("DBI")
library("htmltools")
library("biomaRt")
library("tximport")
```
# Purpose of this homework

The purpose of this homework is to present a clear and complete overview of the datasets you will be analysing for your project, which is useful to identify and solve problems early on. Moreover, part of the results can be re-used for your executive summary. Note that results presented here are not necessarily final: you may still opt to perform modifications to your preprocessing pipeline, design (samples included) etc. for your final report (i.e. presentation & executive summary). 

For this assignment we ask you to do the following:

- Present some general info on your datasets
- Present quality control (QC) on your raw sequencing files
- Load in the data and present the intensity/count/pseudocount data

If you prefer to align your sequencing data yourself (which is highly appreciated), you will already need to do so for this homework. However, since you already have to present QC results for the raw sequencing data, alignment is only a small additional effort (highly similar to what we performed in the practical session).

In order to complete this task you can find a R markdown template via Ufora and also a filled out example. Upon completion of filling out the template, you can upload a knitted html file (best saved as a zip file in order to preserve html structure) via the assignment functionality on Ufora. **Note that one submission per group is sufficient!**

# Project assignment

You can find the project assignment via Ufora.

# Group info

Group number: 

Group members: - - -

Group topic: 

We used the complete datasets/a subset of the datasets (in case of the latter explain why!)


```{r }
table_general_info <- rbind(c("GSE50772","GSE122459"),
                            c(81,26),
                            c(0,0))
colnames(table_general_info) <- c("Microarray Data","RNAseq Data")
rownames(table_general_info) <- c("Accesion ID","Number of Samples (original)",
                                  "Number of Samples (analysed)")

# Link to data Microarray
# https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50772
# Link to data RNAseq
# https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122459

print(table_general_info)
```

# Microarray data

## General info

Microarray type: Affymetrix Human Genome U133 Plus 2.0 Array

## Intensity values

Read in your microarray data and display the head and dimensions of the intensity value matrix.

```{r load-data-raw, message=TRUE}
library(oligo)

# Set the path to the folder containing all CEL filesL
cel_dir <- "C:/Users/Emmanuel/OneDrive - UGent/Documents/Genome analysis/HW2/GSE50772_RAW/GSE50772_RAW"
celfiles <- list.celfiles(cel_dir, full.names = TRUE)

# Load the raw microarray data
data.raw <- read.celfiles(celfiles)

# Preview the first few rows of raw expression values
dim(exprs(data.raw))
head(exprs(data.raw))
```

## Sample annotation

Provide basic sample annotation, including the phenotype of interest and relevant other features (e.g. cofounders). Note that this statement is not final, if during analysis you would drop a cofounder, comparison or add one this is no problem, just try to define an initial goal of your analysis.

```{r}
BiocManager::install("GEOquery")
library(GEOquery)
geo <- getGEO(GEO = "GSE50772")[[1]]
pheno <- pData(geo)
head(pheno)
```

# Sequencing data

## General info

Read length: 

Single/paired end sequencing: Illumina HiSeq 2500 and Illumina NextSeq 500 (! batch effect)

Platform used: 

## Quality control and mapping statistics

Add the multiqc report using `htmltools::includeHTML`. Hint: you can remove the navigation bar on the left by opening the html file > closing the bar > saving the html file. This will remove the indentation in your final html file. In case you start from count data, we still ask you to download (a representative part) of the fastq data to obtain a first impression of the data quality.

```{r, echo=FALSE}

```

If you did not perform (pseudo-)alignment yourself, you can include only the quality control data in the multiqc report

## Sample annotation

Provide basic sample annotation, including the phenotype of interest and relevant other features (e.g. cofounders). Note that this statement is not final, if during analysis you would drop a cofounder, comparison or add one this is no problem, just try to define an initial goal of your analysis.

```{r}
# Creating metadata for 26 samples with GSM IDs
MetaData_RNA <- matrix(nrow = 26, ncol = 1)

rownames(MetaData_RNA) <- c(
  "GSM3466767", "GSM3466768", "GSM3466769", "GSM3466770", "GSM3466771",
  "GSM3466772", "GSM3466773", "GSM3466774", "GSM3466775", "GSM3466776",
  "GSM3466777", "GSM3466778", "GSM3466779", "GSM3466780", "GSM3466781",
  "GSM3466782", "GSM3466783", "GSM3466784", "GSM3466785", "GSM3466786",
  "GSM3466787", "GSM3466788", "GSM3466789", "GSM3466790", "GSM3466791", "GSM3466792"
)

colnames(MetaData_RNA) <- c("condition")

MetaData_RNA[1,1] <- "SLE"
MetaData_RNA[2,1] <- "SLE"
MetaData_RNA[3,1] <- "SLE"
MetaData_RNA[4,1] <- "SLE"
MetaData_RNA[5,1] <- "SLE"
MetaData_RNA[6,1] <- "SLE"
MetaData_RNA[7,1] <- "SLE"
MetaData_RNA[8,1] <- "SLE"
MetaData_RNA[9,1] <- "SLE"
MetaData_RNA[10,1] <- "SLE"
MetaData_RNA[11,1] <- "SLE"
MetaData_RNA[12,1] <- "SLE"
MetaData_RNA[13,1] <- "SLE"
MetaData_RNA[14,1] <- "SLE"
MetaData_RNA[15,1] <- "SLE"
MetaData_RNA[16,1] <- "SLE"
MetaData_RNA[17,1] <- "SLE"
MetaData_RNA[18,1] <- "SLE"
MetaData_RNA[19,1] <- "SLE"
MetaData_RNA[20,1] <- "SLE"
MetaData_RNA[21,1] <- "healthy"
MetaData_RNA[22,1] <- "healthy"
MetaData_RNA[23,1] <- "healthy"
MetaData_RNA[24,1] <- "healthy"
MetaData_RNA[25,1] <- "healthy"
MetaData_RNA[26,1] <- "healthy"

print(MetaData_RNA)
```

## (Pseudo)Count table

Read in your count data or combine your kallisto output to combine your pseudocount data (via tximport) and display the head and dimensions of your count data object.

```{r}

sample_names <- list.files("quants")
n <- length(sample_names)

if (n >= 2) {
  half <- floor(n / 2)
  cond <- c(rep("control", half), rep("treated", n - half))
} else {
  cond <- rep("control", n)
}

samples <- data.frame(sample = sample_names, condition = cond)

files <- file.path("quants", samples$sample, "abundance.tsv")
names(files) <- samples$sample

# Mapping transcript â†’ gene
mart <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")
tx2gene <- getBM(attributes = c("ensembl_transcript_id", "ensembl_gene_id"), mart = mart)
colnames(tx2gene) <- c("TXNAME", "GENEID")

# Import kallisto
txi <- tximport(files, type = "kallisto", tx2gene = tx2gene, ignoreTxVersion = TRUE)

if (length(unique(samples$condition)) > 1) {
  dds <- DESeqDataSetFromTximport(txi, colData = samples, design = ~ condition)
} else {
  dds <- DESeqDataSetFromTximport(txi, colData = samples, design = ~ 1)
}

counts_raw <- counts(dds)
write.csv(counts_raw, file = "raw_count_table.csv")

```

